src/components/AttachmentModal.js

```diff
@@ -330,7 +337,13 @@ function AttachmentModal(props) {
                     setShouldLoadAttachment(true);
                 }}
                 onModalHide={(e) => {
-                    props.onModalHide(e);
+                    props.onModalHide(e).then(() => {
+                        if (!onModalHideCallbackRef.current && !_.isFunction(onModalHideCallbackRef.current)) {
```

Comment:
I think you meant to use OR here not AND. But the second condition is not needed, there should be no case where that ref is set to anything but a function.

Reply:
Update the conditional logic. Thanks! I'll keep the promise for now to avoid the double navigation bug.